### PR TITLE
accelerate tests

### PR DESCRIPTION
**Pull request by Hecate2**
Tests that are usually not related to core compiler code changes are grouped in namespace Peripheral, in order to accelerate the core tests. Description in coverage is refactored to avoid the slow execution of `instruction.Operand.Span.TryGetString(out var str)`

### CODE REVIEW

**Comment by Hecate2** [comment]
Tests that are usually not related to core compiler code changes are grouped in namespace Peripheral, in order to accelerate the core tests. Description in coverage is refactored to avoid the slow execution of `instruction.Operand.Span.TryGetString(out var str)`

File: src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs [hunk 1]
```diff
@@ -24,7 +26,16 @@ public class CoverageHit(int offset, string description, bool outOfScript = fals
         /// <summary>
         /// The instruction description
         /// </summary>
-        public string Description { get; } = description;
+        private string _description { get; } = description;
+        public string Description
```
Comment (shargon) [inline src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs:30]:
```suggestion
        private string _description { get; } = description;

        public string Description
```

**Review by shargon — COMMENTED** [review]
We lose information in `DescriptionFromInstruction`, maybe it's better to avoid this call during test?

**Comment by Hecate2** [comment]
> We lose information in `DescriptionFromInstruction`, maybe it's better to avoid this call during test?

My changes are exactly doing so

**Comment by shargon** [comment]
I mean the method is public and return less information

**Comment by Hecate2** [comment]
Controlling decoding with `bool tryDecodeOperand`

File: src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs [hunk 2]
```diff
@@ -24,7 +31,15 @@ public class CoverageHit(int offset, string description, bool outOfScript = fals
         /// <summary>
         /// The instruction description
         /// </summary>
-        public string Description { get; } = description;
+        public string Description
```
Comment (shargon) [inline src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs:34]: Now we don't need this changes? Isn't it?

Comment (Hecate2) [reply src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs:34]: > Now we don't need this changes? Isn't it?

I am not sure. `[DebuggerDisplay("Offset:{Offset}, Description:{Description}...` still needs additional description with UTF-8 decoded string

File: src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs [hunk 3]
```diff
@@ -86,7 +86,7 @@ internal void GenerateMethods(MethodDetectionMechanism mechanism, ContractState
 
                 if (!_lines.ContainsKey(ip))
                 {
-                    AddLine(instruction, new CoverageHit(ip, CoverageHit.DescriptionFromInstruction(instruction, state.Nef.Tokens), false));
+                    AddLine(instruction, new CoverageHit(ip, CoverageHit.DescriptionFromInstruction(instruction, tryDecodeOperand: false, state.Nef.Tokens), instruction, false));
```
Comment (shargon) [inline src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs:89]: But we are removing the info from coverage, not only for test, isn't it?

Comment (Hecate2) [reply src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs:89]: Again I am not sure😰

Comment (Hecate2) [reply src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs:89]: > But we are removing the info from coverage, not only for test, isn't it?

What do we actually want in nef coverage report and in tests? I do not find the operand decoding useful, except in debugging

**Comment by Hecate2** [comment]
Github actions are having weird problems at checkout 😰

**Review by Jim8y — DISMISSED** [review]
I am not familier with the coverage hit thing, others look good to me.

**Comment by Hecate2** [comment]
This version should be good enough

File: src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs [hunk 4]
```diff
@@ -24,7 +30,12 @@ public class CoverageHit(int offset, string description, bool outOfScript = fals
         /// <summary>
         /// The instruction description
         /// </summary>
-        public string Description { get; } = description;
+        public string Description { get => DescriptionFromInstruction(Instruction, MethodTokens); }
```
Comment (shargon) [inline src/Neo.SmartContract.Testing/Coverage/CoverageHit.cs:33]:
```suggestion
        public string Description => DescriptionFromInstruction(Instruction, MethodTokens);
```

File: src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs [hunk 5]
```diff
@@ -86,7 +89,7 @@ internal void GenerateMethods(MethodDetectionMechanism mechanism, ContractState
 
                 if (!_lines.ContainsKey(ip))
                 {
-                    AddLine(instruction, new CoverageHit(ip, CoverageHit.DescriptionFromInstruction(instruction, state.Nef.Tokens), false));
+                    AddLine(instruction, new CoverageHit(ip, instruction, false, instruction.OpCode == OpCode.CALLT ? state.Nef.Tokens : null));
```
Comment (shargon) [inline src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs:92]: seems that if we send always `Tokens` it will be the same, isn't it?

**Review by shargon — APPROVED** [review]

File: src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs (hunk 5 above)
Comment (Hecate2) [reply src/Neo.SmartContract.Testing/Coverage/CoveredContract.cs:92]: Yes I keep sending the same tokens

**Pull request merged**